core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/response/SdkErrorResponseHandler.java

```diff
@@ -51,6 +51,7 @@ private SdkException handleServiceException(SdkHttpFullResponse response, Execut
         try {
             return delegate.handle(response, executionAttributes);
         } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
             throw e;
```

Comment:
Same question here. Where were we not interrupting the thread before? Can we just fix it there instead of re-interrupting here?